src/Components/PostInfo.tsx

```diff
@@ -0,0 +1,20 @@
+import { CommentList } from './CommentList';
+import { UserInfo } from './UserInfo';
+
+export const PostInfo: React.FC<Post> = ({
+  title,
+  body,
+  user,
+  comments,
+}) => (
+  <div className="ui piled segments">
+    <div className="ui segment">
+      <h1 className="header">{title}</h1>
+      <p className="description">{body}</p>
+      {user && <UserInfo {...user} />}
+    </div>
+    <div className="ui secondary segment">
+      {comments && <CommentList comments={comments} />}
```

Comment:
comments will always be an array (empty or non-empty), that always be true. You need another check here